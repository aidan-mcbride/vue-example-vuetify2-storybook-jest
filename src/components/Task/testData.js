export const taskData = {
  id: '1',
  title: 'Test Task',
  state: 'TASK_INBOX',
  updated_at: new Date(2019, 0, 1, 9, 0)
};
