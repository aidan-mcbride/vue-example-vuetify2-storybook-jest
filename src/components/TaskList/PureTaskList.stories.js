import PureTaskList from './PureTaskList';

import { defaultTasksData, withPinnedTasksData } from './testData';
import { actionsData } from '@/components/Task/Task.stories';

const paddedList = () => {
  return {
    template:
      '<div style="padding: 3rem; background-color: #CFD8DC"><story/></div>'
  };
};

export default {
  title: 'TaskList',
  excludeStories: /.*Data$/,
  decorators: [paddedList]
};

// default TaskList state
export const Default = () => ({
  components: { PureTaskList },
  template: `<pure-task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  props: {
    tasks: {
      default: () => defaultTasksData
    }
  },
  methods: actionsData
});

// TaskList with pinned tasks
export const WithPinnedTasks = () => ({
  components: { PureTaskList },
  template: `<pure-task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  props: {
    tasks: {
      default: () => withPinnedTasksData
    }
  },
  methods: actionsData
});

// loading TaskList state
export const Loading = () => ({
  components: { PureTaskList },
  template: `<pure-task-list loading @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  methods: actionsData
});

// empty TaskList state
export const Empty = () => ({
  components: { PureTaskList },
  template: `<pure-task-list @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  methods: actionsData
});
