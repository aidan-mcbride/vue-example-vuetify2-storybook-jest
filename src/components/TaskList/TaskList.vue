<template>
  <div>
    <v-list v-if="showTasks">
      <task
        v-for="(task, index) in tasks"
        :key="index"
        :task="task"
        @archiveTask="$emit('archiveTask', $event)"
        @pinTask="$emit('pinTask', $event)"
      />
    </v-list>
  </div>
</template>

<script>
import Task from '@/components/Task/Task';
export default {
  name: 'TaskList',
  components: {
    Task
  },
  props: {
    loading: {
      type: Boolean,
      default: false
    },
    tasks: {
      type: Array,
      default: () => []
    }
  },
  computed: {
    noTasks() {
      return this.tasks.length === 0;
    },
    showTasks() {
      return !this.loading && !this.noTasks;
    }
  }
};
</script>

<style lang="scss" scoped></style>
