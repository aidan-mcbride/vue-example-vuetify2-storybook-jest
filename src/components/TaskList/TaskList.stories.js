import TaskList from './TaskList.vue';

import { defaultTasksData } from './testData';
import { actionsData } from '@/components/Task/Task.stories';

const paddedList = () => {
  return {
    template: '<div style="padding: 3rem;"><story/></div>'
  };
};

export default {
  title: 'TaskList',
  excludeStories: /.*Data$/,
  decorators: [paddedList]
};

// default TaskList state
export const Default = () => ({
  components: { TaskList },
  template: `<task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  props: {
    tasks: {
      default: () => defaultTasksData
    }
  },
  methods: actionsData
});
