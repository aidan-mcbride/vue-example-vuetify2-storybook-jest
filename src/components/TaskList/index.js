import TaskList from './TaskList.vue';
import BaseTaskList from './BaseTaskList.vue';

export { TaskList, BaseTaskList };
