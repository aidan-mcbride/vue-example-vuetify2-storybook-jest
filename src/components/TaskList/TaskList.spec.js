// Components
import TaskList from './TaskList';

// Utilities
import { mount } from '@vue/test-utils';
import { defaultTasksData } from './testData';

describe('TaskList', () => {
  describe('Default State', () => {
    it('renders correct markup', () => {
      const wrapper = mount(TaskList, {
        propsData: {
          tasks: {
            default: () => defaultTasksData
          }
        }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
