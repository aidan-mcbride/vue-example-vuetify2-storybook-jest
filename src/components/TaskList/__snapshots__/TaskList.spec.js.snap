// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Default State renders correct markup 1`] = `<div tasks="[object Object]"></div>`;
